src/SGFTree.cpp

```diff
@@ -449,9 +449,7 @@ std::string SGFTree::state_to_string(GameState& pstate, int compcolor) {
 
     while (state->forward_move()) {
         int move = state->get_last_move();
-        if (move == FastBoard::RESIGN) {
-            break;
-        }
+        assert(move != FastBoard::RESIGN) {
```

Comment:
Here I do not know what you want to do. Is assert meant as an if or you just forgot the semicolon?